src/main/java/roomescape/reservation/presentation/controller/ReservationController.java

```diff
@@ -0,0 +1,57 @@
+package roomescape.reservation.presentation.controller;
+
+import jakarta.validation.Valid;
+import java.net.URI;
+import java.util.List;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import roomescape.auth.dto.LoginMember;
+import roomescape.auth.presentation.AuthenticatedMember;
+import roomescape.reservation.application.ReservationQueryService;
+import roomescape.reservation.application.ReservationService;
+import roomescape.reservation.application.dto.ReservationDto;
+import roomescape.reservation.infrastructure.dto.ReservationDetailData;
+import roomescape.reservation.presentation.dto.request.ReservationRequest;
+import roomescape.reservation.presentation.dto.response.ReservationDetailResponse;
+
+@RestController
+@RequestMapping("/reservations")
+public class ReservationController {
+
+    private final ReservationService service;
+    private final ReservationQueryService queryService;
+
+    public ReservationController(ReservationService service, ReservationQueryService queryService) {
+        this.service = service;
+        this.queryService = queryService;
+    }
+
+    @GetMapping
+    public List<ReservationDetailResponse> getAllReservations(ReservationSearchCondition condition) {
+        List<ReservationDetailData> reservationDetails = queryService.getReservationDetails(condition);
+        return ReservationDetailResponse.from(reservationDetails);
+    }
+
+    @PostMapping
+    public ResponseEntity<ReservationDetailResponse> createReservation(
+            @Valid @RequestBody ReservationRequest request,
+            @AuthenticatedMember LoginMember member) {
```

Comment:
```suggestion
            @AuthenticatedMember LoginMember member) {
```
위 부분에 대해서, 이번에 request.setAttribute 형태로 변경된 김에 Custom Annotation 없이 `@RequestAttribute("loginMember") LoginMember member` 이런 형태로도 사용해볼 수 있을 것 같거든요! 이 부분은 어떻게 생각하시는지 궁금해요!!

Reply:
오  `@RequestAttribute`는 처음 알았어요. HTTP 요청(request)의 attribute에 저장된 값을 컨트롤러 메서드 파라미터로 바인딩할 때 사용하는 어노테이션이군요!

CustomAnnotation과 비교해봤을 때 코드량을 제외한다면 약간의 단점이 있는 것 같습니다.

**인증에 관한 정보를 Controller가 알아야합니다.** 
request에 "loginMember"로 저장되어있다는 것을 알아야 하기 때문입니다. 상수화로 관리해볼 수 있겠지만 Controller가 인증 관련 정보를 알아야한다는 사실 자체가 단점인 것 같습니다.

그리고 개인적으로 인증정보가 "꺼내 쓰는" 느낌보다, "주입 되는" 느낌을 주는 것이 관심사가 더 잘 분리되었다고 느낍니다. 객관적이지 못한 것 같지만, 그만큼 `@RequestAttribute`도 사실상 잘 동작하더군요. 이 둘 중에 선택은 개인적인 선호일까요? 제가 고려하지 못한 장단점이 또 있다면 알려주세요!

Reply:
커스텀 어노테이션을 사용한다면 연결된 리졸버에서 추가적인 로직을 수행할 수 있는 장점도 있을 것 같아요! 그게 필요없다면 지금은 requestAttribute를 써도 상관은 없을 것 같고용ㅎㅎ